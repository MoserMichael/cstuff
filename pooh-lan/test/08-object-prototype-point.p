
sub make_point( x, y )

    return { 'x_pos' : x,
             
             'y_pos' : y,
             
             'move'  : sub ( dx, dy ) 
                            this . x_pos = this . x_pos + dx
                            this . y_pos = this . y_pos + dy

                            return this
                       end,

             'print' : sub ( )
#                           println( ~msg 'point x-pos: [ this . x_pos ]  y-pos: [ this . y_pos ] ' )
                            println( ~msg 'point x-pos: ' .. this . x_pos .. '  y-pos: ' ..  this . y_pos  )
                       end,

             'get_x' : sub ( )
                            return this . x_pos
                       end,

             'get_y' : sub ( )
                            return this . y_pos  
                       end
            }
end            


pt = make_point( ~x 10 ~y 10 )

pt . print ()

pt . move ( ~dx 1 ~dy -1 )

pt . print ()

println( ~msg 'point-accessors x = [ pt . get_x () ]  y = [ pt . get_y() ]' )

             
