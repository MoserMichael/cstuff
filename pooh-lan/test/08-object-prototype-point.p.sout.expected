point x-pos: 10  y-pos: 10
point x-pos: 11  y-pos: 9
point-accessors x = 11  y = 9
